applicationset/generators/pull_request.go

```diff
@@ -118,10 +120,18 @@ func (g *PullRequestGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha
 // selectServiceProvider selects the provider to get pull requests from the configuration
 func (g *PullRequestGenerator) selectServiceProvider(ctx context.Context, generatorConfig *argoprojiov1alpha1.PullRequestGenerator, applicationSetInfo *argoprojiov1alpha1.ApplicationSet) (pullrequest.PullRequestService, error) {
 	if generatorConfig.Github != nil {
+		err := ScmProviderAllowed(generatorConfig.Github.API, g.allowedSCMProviders)
```

Comment:
It's a bit verbose, but could we also log a warning?

```go
	log.WithFields(log.Fields{
		common.SecurityField:    common.SecurityMedium,
                 "applicationset": applicationSetInfo.Name,
                 "appSetNamespace": applicationSetInfo.Namespace,
	}).Debugf("attempted to use disallowed SCM %q: %v", generatorConfig.Github.API, err)
```

Reply:
Maybe to avoid duplication, we could throw a new `ErrDisallowedSCM` and then handle/log in the `selectServiceProvider` caller.